terraform/groups/lambda/main.tf

```diff
@@ -25,9 +28,14 @@ data "terraform_remote_state" "network_remote_state" {
   }
 }
 
+data "vault_generic_secret" "aws_account_id" {
```

Comment:
That was my first attempt, but I got a kind-of circular-reference error.
I think it's because it's used in the assume_role ARN in the provider:
```
provider "aws" {
  assume_role {
    role_arn = "arn:aws:iam::${local.aws_account_id}:role/payment-reconciler-provisioning-role"
  }
}
```
😖
I agree it would be better though.
I'll give it another go though, see if I can get it to work.

Reply:
Yeah, tried this again and it doesn't work. The error is:
`Error: Cycle: data.aws_caller_identity.current, local.aws_account_id, provider.aws`
...but I can't find a lot in the Terraform docs about it.

Best explanation I could find is [here](https://serverfault.com/questions/1005761/what-does-error-cycle-means-in-terraform).
Tried the suggested `terraform graph -draw-cycles` command and it gives:
![Screen Shot 2020-05-20 at 14 27 52](https://user-images.githubusercontent.com/53176667/82451636-2d17bb00-9aa6-11ea-9ccc-27ada82e136f.png)